src/NexusMods.App.UI/MessageBox/MessageBoxViewModel.cs

```diff
@@ -0,0 +1,189 @@
+using System.Reactive;
+using Avalonia.Threading;
+using ExCSS;
+using NexusMods.App.UI.MessageBox.Enums;
+using ReactiveUI;
+
+namespace NexusMods.App.UI.MessageBox;
+
+public class MessageBoxViewModel : IMessageBoxViewModel<ButtonResult>
+{
+    public readonly ClickEnum _defaultEnterButton;
+    public readonly ClickEnum _defaultEscapeButton;
+    private IMessageBoxView<ButtonResult>? _view;
+
+    public string ContentTitle { get; }
+    public string ContentMessage { get; set; }
+
+    public bool IsOkShowed { get; private set; }
+    public bool IsYesShowed { get; private set; }
+    public bool IsNoShowed { get; private set; }
+    public bool IsAbortShowed { get; private set; }
+    public bool IsCancelShowed { get; private set; }
```

Comment:
A list of buttons should be generated and shown based on the inputs rather than having a limited set ready. 

I'd expect button creation and handling to happen in the codebehind. 
VM would have SetResult(buttonId) to be called when a button in pressed.